### PR TITLE
PDA cart swapping, eject hotkey and verb; toggle light action button

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

- If you click your PDA with a cartridge and it already has one, you'll automatically eject it before putting the new one in 
- You can also eject the cartridge without opening the UI by using ctrl+shift+click or the Remove Cartridge verb
- PDAs now have a toggle light action button, like other light sources
- PDA UI correctly updates when ejecting or inserting a cartridge or ejecting an ID

## Why It's Good For The Game

This has bugged me for years, so I'm assuming it bugs other people too.

## Changelog
:cl: cacogen
add: Ctrl+shift+click PDA to eject cartridge
add: Clicking PDA with cartridge swaps out current cartridge
add: Remove Cartridge verb for people using the right-click menu or the Object tab
add: PDA toggle light action button
fix: PDA UI now updates when ejecting or inserting cartridge or ejecting ID
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
finally

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -807,6 +815,18 @@ GLOBAL_LIST_EMPTY(PDAs)
 	else
 		to_chat(usr, "<span class='warning'>This PDA does not have a pen in it!</span>")
 
+/obj/item/pda/proc/eject_cart()
+	if(issilicon(usr) || !usr.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
```
Comment (Cyberboss) [inline code/game/objects/items/devices/PDA/PDA.dm]: Think telekenetics should apply here, no?

Comment (Cyberboss) [inline code/game/objects/items/devices/PDA/PDA.dm]: This proc shouldn't use `usr` and should take a `mob/user` parameter.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (cacogen) [reply code/game/objects/items/devices/PDA/PDA.dm]: TK doesn't work with the ID or pen and letting it work with the cart makes it teleport into your hand. I'm too lazy to add a check to make it drop like IDs in computers.

**Comment by cacogen** [comment]
This should be good to go if you want to review it again @cyberboss

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**